x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceNamedWriteablesProvider.java

```diff
@@ -98,6 +98,16 @@ public static List<NamedWriteableRegistry.Entry> getNamedWriteables() {
         namedWriteables.add(
             new NamedWriteableRegistry.Entry(ServiceSettings.class, CohereServiceSettings.NAME, CohereServiceSettings::new)
         );
+        namedWriteables.add(
```

Comment:
I'm keeping `CohereServiceSettings` it's own class because I think we'll use it again for other inference actions for cohere. The way I structured `CohereEmbeddingsServiceSettings` was to make the common settings be in `CohereServiceSettings` (e.g. the model name, dimensions etc) and put the `embedding_type` field within `CohereEmbeddingsServiceSettings`.